### PR TITLE
Add telegraf-operator chart

### PR DESCRIPTION
**Pull request by gitirabassi**
Closes https://github.com/influxdata/telegraf-operator/issues/7

This will need https://github.com/influxdata/telegraf-operator/milestone/2 to be completed

I need helm configuring this to be proper helm etiquette

What is the proper way of adding a secret with helm? @naseemkullah 

I'm thinking of adding `influxdb` as a dependency and then adding the secret to send telegraf data to influxdb instance. 

This can't be done with influxdbV2 because of all the initialization and token distribution

### CODE REVIEW

**Comment by rawkode** [comment]
Dependencies in Helm are pretty shit. The secret should be provided upfront via a value `secretName` `

**Comment by nsteinmetz** [comment]
Please don't:

> I'm thinking of adding influxdb as a dependency and then adding the secret to send telegraf data to influxdb instance.

InfluxDB instance can be elsewhere or one may want to user another output system than InfluxDB

In my case, my monitoring system is hosted externally from the kubernetes cluster ; I don't want to deploy a influxdb instance which I'll not use.

**Comment by gitirabassi** [comment]
@nsteinmetz Thank you for your feedback. I'm all for customizability but also making sure that people that are getting started find everything they need here.

Just like it's possible to disable the `cert-manager`  dependency now, I'll make `influxdb` dependency optional. 

The question now is: 
- should it be enabled by default
- should it be disabled by default
cc @rawkode

**Comment by rawkode** [comment]
@gitirabassi Disabled by default. I don't like coupling, and we'll want to support the InfluxDB Operator later - so lets not tie this to InfluxDB too much now.

**Comment by gitirabassi** [comment]
@rawkode can you help me understand what is going on with CI ? I can't figure out why it's failing

File: charts/telegraf-operator/README.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+helm repo add influxdata https://helm.influxdata.com
+
+helm install telegraf-operator influxdata/telegraf-operator
+
+```
+
+## Contributing
+
+```
+helm dependency update
```
Comment (rawkode) [inline charts/telegraf-operator/README.md:17]: Is this required?

File: charts/telegraf-operator/templates/certificate.yml [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  namespace: {{ .Values.namespace }}
+spec:
+  dnsNames:
+  - "{{ include "telegraf-operator.fullname" . }}.{{ .Values.namespace }}.svc"
+  isCA: true
+  issuerRef:
+    kind: Issuer
+    name: selfsigned
+    namespace: {{ .Values.namespace }}
+  secretName: webhook-server-cert
```
Comment (rawkode) [inline charts/telegraf-operator/templates/certificate.yml:17]: Should this be something like `telegraf-operator-tls` ?

File: charts/telegraf-operator/templates/deployment.yaml [hunk 3]
```diff
@@ -0,0 +1,72 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ include "telegraf-operator.fullname" . }}
+  namespace: {{ .Values.namespace }}
+  labels:
+    {{- include "telegraf-operator.labels" . | nindent 4 }}
+spec:
+  replicas: {{ .Values.replicaCount }}
```
Comment (rawkode) [inline charts/telegraf-operator/templates/deployment.yaml:9]: Can the operator be horizontally scaled, or should we hard code this to 0?

File: charts/telegraf-operator/templates/issuer.yml [hunk 4]
```diff
@@ -0,0 +1,11 @@
+{{- if .Values.certManager.certificate.enable -}}
+apiVersion: cert-manager.io/v1alpha2
+kind: Issuer
+metadata:
+  name: selfsigned
```
Comment (rawkode) [inline charts/telegraf-operator/templates/issuer.yml:5]: Should this be a `telegraf-operator-issuer`?

Comment (gitirabassi) [reply charts/telegraf-operator/templates/issuer.yml:5]: done

File: charts/telegraf-operator/templates/deployment.yaml (hunk 3 above)
Comment (gitirabassi) [reply charts/telegraf-operator/templates/deployment.yaml:9]: it already defaults to 3

File: charts/telegraf-operator/templates/certificate.yml (hunk 2 above)
Comment (gitirabassi) [reply charts/telegraf-operator/templates/certificate.yml:17]: done

File: charts/telegraf-operator/README.md (hunk 1 above)
Comment (gitirabassi) [reply charts/telegraf-operator/README.md]: done

**Comment by gitirabassi** [comment]
@rawkode This is now ready for final review

**Pull request merged**